tron/config/config_parse.py

```diff
@@ -270,6 +273,68 @@ class ValidateSecretSource(Validator):
 valid_secret_source = ValidateSecretSource()
 
 
+def valid_permission_mode(value: Union[str, int], config_context: ConfigContext) -> str:
+    try:
+        decimal_value = int(
+            str(value), base=8
+        )  # take in permission mode as string or int representation of an octal number. Goes from 0 to 4095 in decimal.
+    except ValueError:
+        error_msg = "Could not parse {} as octal permission mode at {}"
+        raise ConfigError(error_msg.format(value, config_context.path))
+    if decimal_value > 4095 or decimal_value < 0:
+        error_msg = "Octal permission mode {} out of bound at {}"
+        raise ConfigError(error_msg.format(value, config_context.path))
+    return str(value)
+
+
+class ValidateSecretVolumeItem(Validator):
+    config_class = ConfigSecretVolumeItem
+
+    validators = {
+        "key": valid_string,  # name of current secret
+        "path": valid_string,  # New secret filename
+        "mode": valid_permission_mode,  # Octal permission mode
+    }
+
+
+valid_secret_volume_item = ValidateSecretVolumeItem()
+
+
+class ValidateSecretVolume(Validator):
+    config_class = ConfigSecretVolume
+
+    optional = True
+    defaults = {
+        "default_mode": "0644",
+        "items": None,
+    }
+
+    validators = {
+        "container_path": valid_string,
+        "secret_volume_name": valid_string,
+        "secret_name": valid_string,
+        "default_mode": valid_permission_mode,
+        "items": build_list_of_type_validator(valid_secret_volume_item, allow_empty=True),
+    }
+
+    def post_validation(self, valid_input, config_context):
+        """Propagate default mode and enforce the secret-key match."""
+        # Our secrets will really only ever have one key, so weirdly we only care about a single item of this array AND it must have the same name as the secret (which is the single key)
```

Comment:
we really should have picked a different config format for re-naming keys :/ 

although, I guess re-using the k8s format leaves us open to adding multi-key secret support eventually - that is to say, this is less weird and more an unfortunate side-effect of our choices :p